EIPS/eip-4610.md

```diff
@@ -0,0 +1,98 @@
+---
+eip: 4610
+title: ERC-721 Delegator Extension
+description: Enables ERC-721 to be loaned
+author: AFKDAO (@AFKDAO)
+discussions-to: https://ethereum-magicians.org/t/delegator-extension-that-enables-erc721-to-be-loaned/7885
+status: Draft
+type: Standards Track
+category: ERC
+created: 2021-12-28
+requires: 721
+---
+
+## Abstract
+The optional ERC-721 Delegator Extension provides a standardized function which could enable ERC-721 to be loaned. 
+
+## Motivation
+The ERC-721 standard does not realize the separation of ownership and use rights, which makes it difficult to realize the loan of ERC-721. Solving this problem will help make ERC-721 more widely used.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+**Every ERC-4610 compliant contract must implement the `ERC721` interfaces** (subject to "caveats" below):
+
+```solidity
+pragma solidity ^0.8.0;
+
+import "./IERC721.sol";
+
+interface IERC4610 is IERC721 {
+
+    /**
+     * @dev Emitted when `owner` or `approved account` enables `setDelegator` to manage the `tokenId` token.
+     */
+    event SetDelegator(address indexed caller, address indexed delegator, uint256 indexed tokenId);
+
```

Comment:
I think there should probably be a view function returning a bool or address to see who's a delegator on a particular NFT.

Reply:
Yes, there is this view function, I named it `delegatorOf`. You can see the current implementation of eip-4610 here： https://github.com/AFKDAO/ERC4610/blob/main/contracts/ERC4610.sol